libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/BatchingSyncUpTarget.java

```diff
@@ -0,0 +1,228 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.smartsync.target;
+
+import com.salesforce.androidsdk.rest.RestRequest;
+import com.salesforce.androidsdk.rest.RestResponse;
+import com.salesforce.androidsdk.smartstore.store.SmartStore;
+import com.salesforce.androidsdk.smartsync.manager.SyncManager;
+import com.salesforce.androidsdk.smartsync.util.Constants;
+import com.salesforce.androidsdk.smartsync.util.SyncState;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Subclass of SyncUpTarget that batches create/update/delete operations by using composite api
+ */
+public class BatchingSyncUpTarget extends SyncUpTarget implements AdvancedSyncUpTarget {
+
+    // Constants
+    public static final String SOUP_NAME = "soupName";
+    public static final String MAX_BATCH_SIZE = "maxBatchSize";
+
+    // Soup records are coming from
+    protected String soupName;
+
+    // Max batch size
+    protected int maxBatchSize;
+
+    /**
+     * Construct SyncUpTarget
+     */
+    public BatchingSyncUpTarget(List<String> createFieldlist, List<String> updateFieldlist, String soupName, int maxBatchSize) {
+        super(createFieldlist, updateFieldlist);
+        this.soupName = soupName;
+        this.maxBatchSize = Math.min(maxBatchSize, 25); // composite api allows up to 25 subrequests
+    }
+
+    /**
+     * Construct SyncUpTarget from json
+     * @param target
+     * @throws JSONException
+     */
+    public BatchingSyncUpTarget(JSONObject target) throws JSONException {
+        super(target);
+        this.soupName = target.getString(SOUP_NAME);
+        this.maxBatchSize = target.getInt(MAX_BATCH_SIZE);
```

Comment:
This should also have a check for max of 25 like the constructor above, i.e. `Math.min(...)`.